src/Sarif/Writers/BaseLogger.cs

```diff
@@ -53,13 +53,14 @@ public bool ShouldLog(Notification notification)
             return _failureLevels.Contains(notification.Level);
         }
 
-        public bool ShouldLog(Result result)
+        public bool ShouldLog(Result result, ReportingDescriptor rule)
```

Comment:
>ShouldLog [](http://example.com/codeflow?start=20&length=9)

this is another existing issue, 
Shouldlog() does not consider the default setting.